src/Init/Data/BitVec/Lemmas.lean

```diff
@@ -852,6 +852,77 @@ protected theorem extractLsb_ofNat (x n : Nat) (hi lo : Nat) :
     (extractLsb' start len x).getLsbD i = (i < len && x.getLsbD (start+i)) := by
   simp [getLsbD, Nat.lt_succ]
 
+/-
+Get the most significant bit after `extractLsb'`. With `extractLsb'`, we extract
+a `BitVec` `x'` with length `len` from `BitVec` `x`, starting from the element
+at position `start`. The function `getMsb` extracts a bit counting from the most
+significant bit. Assuming certain conditions, `x.(extractLsbD' {w} start len).getMsbD i`
+is equal to `x.getMsbD (w - (start + len - i))`.
+
+Example (w = 9, start := 3, len := 4):
+
+                                |---| = w - (start+len) = 3
+                                      |start + len|     = 7
+                                            |start|     = 4
+                                      |len|             = 3
+let x                       =   9 8 7 6 5 4 3 2 1 0
+let x' = x.extractLsb' 3 4  =       7 6 5 4
+                              |       |
+                              x.getMsbD 0
+                                      | x.getMsbD (i := 4) =
+                                        x'.getMsbD (i := 1) =
+                                        x.getMsbD (i := w - (start + len - i) =
+                                                        10 - (4 + 3 - 1) = 4)
+# Condition 1: `i < len`
+
+The index `i` must be within the range of `len`.
+
+# Condition 2: `start + len - i ≤ w`
+
+If `start + len` is larger than `w`, some lower locations `i` in `x'` are
```

Comment:
```suggestion
If `start + len` is larger than `w`, the high bits `i` in `x'` are
```

I find "lower" to be ambiguous, since this depends on whether one is thinking in terms of `msb` or `lsb`. I don't have *great* suggestions here, but I at least prefer `rightward`, or perhaps `high bits`.